src/designspace.rs

```diff
@@ -0,0 +1,277 @@
+    /// Corresponds with font.info.postscriptFontName
+    pub postscriptfontname: Option<String>,
+    /// Corresponds with styleMapFamilyName
+    pub stylemapfamilyname: Option<String>,
+    /// Corresponds with styleMapStyleName
+    pub stylemapstylename: Option<String>,
+    /// Location in designspace.
+    pub location: Vec<Dimension>,
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#instance-element
+#[derive(Deserialize)]
+struct InstanceXmlRepr {
+    pub familyname: Option<String>,
+    pub stylename: Option<String>,
+    pub name: String,
+    pub filename: Option<String>,
+    pub postscriptfontname: Option<String>,
+    pub stylemapfamilyname: Option<String>,
+    pub stylemapstylename: Option<String>,
+    pub location: LocationXmlRepr,
+}
+
+impl From<InstanceXmlRepr> for Instance {
+    fn from(instance_xml: InstanceXmlRepr) -> Self {
+        Instance {
+            familyname: instance_xml.familyname,
+            stylename: instance_xml.stylename,
+            name: instance_xml.name,
+            filename: instance_xml.filename,
+            postscriptfontname: instance_xml.postscriptfontname,
+            stylemapfamilyname: instance_xml.stylemapfamilyname,
+            stylemapstylename: instance_xml.stylemapstylename,
+            location: instance_xml.location.dimension,
+        }
+    }
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#location-element-top-level-stat-label
+#[derive(Deserialize)]
+struct LocationXmlRepr {
+    pub dimension: Vec<Dimension>,
+}
+
+/// A [design space dimension]].
+///
+/// [design space location]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#location-element-source
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+pub struct Dimension {
+    /// Name of the axis, e.g. Weight.
+    pub name: String,
+    /// Value on the axis in user coordinates.
+    pub uservalue: Option<f32>,
+    /// Value on the axis in designcoordinates.
+    pub xvalue: Option<f32>,
+    /// Separate value for anisotropic interpolations.
+    pub yvalue: Option<f32>,
+}
+
+impl DesignSpaceDocument {
+    /// Load a designspace.
+    pub fn load<P: AsRef<Path>>(path: P) -> Result<DesignSpaceDocument, DesignSpaceLoadError> {
+        let reader = BufReader::new(File::open(path).map_err(DesignSpaceLoadError::Io)?);
+        quick_xml::de::from_reader(reader).map_err(DesignSpaceLoadError::DeError)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::Path;
+
+    use pretty_assertions::assert_eq;
+
+    use crate::designspace::{AxisMapping, Dimension};
+
+    use super::DesignSpaceDocument;
+
+    fn dim_name_xvalue(name: &str, xvalue: f32) -> Dimension {
+        Dimension { name: name.to_string(), uservalue: None, xvalue: Some(xvalue), yvalue: None }
+    }
+
+    #[test]
+    fn read_single_wght() {
+        let ds = DesignSpaceDocument::load(Path::new("testdata/single_wght.designspace")).unwrap();
```

Comment:
Just a note that AsRef<Path> is implemented for &str so the Path::new is unnecessary here.

Reply:
Removed the pointless Path::new, ty.